include/flatbuffers/grpc.h

```diff
@@ -23,50 +23,271 @@
 #include "grpc++/support/byte_buffer.h"
 #include "grpc/byte_buffer_reader.h"
 
+namespace flatbuffers {
 namespace grpc {
 
+enum AddRef { ADD_REF };
+enum StealRef { STEAL_REF };
+
+// Message is a typed wrapper around a buffer that manages the underlying
+// `grpc_slice` and also provides flatbuffers-specific helpers such as `Verify`
+// and `GetRoot`. Since it is backed by a `grpc_slice`, the underlying buffer
+// is refcounted and ownership is be managed automatically.
+template <class T>
+class Message {
+ public:
+  inline Message() : slice_(grpc_empty_slice()) {}
+
+  inline Message(grpc_slice slice, AddRef) : slice_(grpc_slice_ref(slice)) {}
```

Comment:
It's both more readable and also less, since it took me a bit of time to figure out why the types were needed :)